src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/NativeMethods.cs

```diff
@@ -472,30 +457,23 @@ RPC_AUTH_IDENTITY_HANDLE AuthIdentity
             PDNS_RECORD *ppQueryResultsSet,
             PVOID *pReserved
             );*/
-        [DllImport(global::Interop.Libraries.Dnsapi, EntryPoint = "DnsQuery_W", CharSet = CharSet.Unicode)]
-        internal static extern int DnsQuery(
-            [In] string recordName,
-            [In] short recordType,
-            [In] int options,
-            [In] IntPtr servers,
-            [Out] out IntPtr dnsResultList,
-            [Out] IntPtr reserved);
+        [GeneratedDllImport(global::Interop.Libraries.Dnsapi, EntryPoint = "DnsQuery_W", CharSet = CharSet.Unicode)]
+        internal static partial int DnsQuery(
+            string recordName,
+            short recordType,
+            int options,
+            IntPtr servers,
+            out IntPtr dnsResultList,
+            IntPtr reserved);
 
         /*VOID WINAPI DnsRecordListFree(
             PDNS_RECORD pRecordList,
             DNS_FREE_TYPE FreeType
             );*/
-        [DllImport(global::Interop.Libraries.Dnsapi, CharSet = CharSet.Unicode)]
-        internal static extern void DnsRecordListFree(
-            [In] IntPtr dnsResultList,
-            [In] bool dnsFreeType);
-
-        /*BOOL GetVersionEx(
-              LPOSVERSIONINFO lpVersionInfo
-            );*/
-        [DllImport(global::Interop.Libraries.Kernel32, EntryPoint = "GetVersionExW", CharSet = CharSet.Unicode, SetLastError = true)]
-        internal static extern bool GetVersionEx(
-                [In, Out] OSVersionInfoEx ver);
+        [GeneratedDllImport(global::Interop.Libraries.Dnsapi, CharSet = CharSet.Unicode)]
```

Comment:
Why is the global:: needed on all of these?

Reply:
Yeah, I had wondered the same thing - there's a `System.DirectoryServices.Interop`, so these all had the `global::`